pkg/kn/commands/revision/describe.go

```diff
@@ -46,19 +62,225 @@ func NewRevisionDescribeCommand(p *commands.KnParams) *cobra.Command {
 				return err
 			}
 
-			printer, err := revisionDescribePrintFlags.ToPrinter()
-			if err != nil {
-				return err
+			if machineReadablePrintFlags.OutputFlagSpecified() {
+				printer, err := machineReadablePrintFlags.ToPrinter()
+				if err != nil {
+					return err
+				}
+				return printer.PrintObj(revision, cmd.OutOrStdout())
 			}
-
-			err = printer.PrintObj(revision, cmd.OutOrStdout())
-			if err != nil {
-				return err
+			printDetails, err := cmd.Flags().GetBool("verbose")
+			var service *v1alpha1.Service
+			serviceName, ok := revision.Labels[servingserving.ServiceLabelKey]
+			if printDetails && ok {
+				service, err = client.GetService(serviceName)
+				if err != nil {
+					return err
+				}
 			}
-			return nil
+			// Do the human-readable printing thing.
+			return describe(cmd.OutOrStdout(), revision, service, printDetails)
 		},
 	}
-	commands.AddNamespaceFlags(revisionDescribeCmd.Flags(), false)
-	revisionDescribePrintFlags.AddFlags(revisionDescribeCmd)
-	return revisionDescribeCmd
+	flags := command.Flags()
+	commands.AddNamespaceFlags(flags, false)
+	machineReadablePrintFlags.AddFlags(command)
+	flags.BoolP("verbose", "v", false, "More output.")
+	return command
+}
+
+func describe(w io.Writer, revision *v1alpha1.Revision, service *v1alpha1.Service, printDetails bool) error {
+	dw := printers.NewPrefixWriter(w)
+	commands.WriteMetadata(dw, &revision.ObjectMeta, printDetails)
+	WriteImage(dw, revision)
+	WritePort(dw, revision)
+	WriteEnv(dw, revision, printDetails)
+	WriteScale(dw, revision)
+	WriteConcurrencyOptions(dw, revision)
+	WriteResources(dw, revision)
+	serviceName, ok := revision.Labels[servingserving.ServiceLabelKey]
+	if ok {
+		serviceSection := dw.WriteAttribute("Service", serviceName)
+		if printDetails {
+			serviceSection.WriteAttribute("Config Gen", revision.Labels[servingserving.ConfigurationGenerationLabelKey])
+			serviceSection.WriteAttribute("Latest Created", strconv.FormatBool(revision.Name == service.Status.LatestCreatedRevisionName))
+			serviceSection.WriteAttribute("Latest Ready", strconv.FormatBool(revision.Name == service.Status.LatestReadyRevisionName))
+			percent, tags := trafficForRevision(revision.Name, service)
+			if percent != 0 {
+				serviceSection.WriteAttribute("Traffic", strconv.FormatInt(int64(percent), 10)+"%")
+			}
+			if len(tags) > 0 {
+				commands.WriteSliceDesc(serviceSection, tags, "Tags", printDetails)
+			}
+		}
+
+	}
+	dw.WriteLine()
+	commands.WriteConditions(dw, revision.Status.Conditions, printDetails)
+	if err := dw.Flush(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func WriteConcurrencyOptions(dw printers.PrefixWriter, revision *v1alpha1.Revision) {
+	target := clientserving.ConcurrencyTarget(&revision.ObjectMeta)
+	limit := revision.Spec.ContainerConcurrency
+	if target != nil || limit != nil {
+		section := dw.WriteAttribute("Concurrency", "")
+		if limit != nil {
```

Comment:
I think we shouldn't write out the limit if its `== 0`, as it is the case by default.

Reply:
That's my output with setting no concurrency options at all:

```
Name:         random-cwvnj-1
Namespace:    default
Age:          25s
Image:        rhuss/random:1.0 (pinned to 946b7c)
Concurrency:
  Limit:      0
Service:      random

Conditions:
  OK TYPE                  AGE REASON
  ++ Ready                 21s
  ++ ContainerHealthy      21s
  ++ ResourcesAvailable    21s
  ++ Active                21s
```